crates/wasmparser/src/readers/core/operators.rs

```diff
@@ -1176,3 +1187,1280 @@ impl<'a> Iterator for OperatorsIteratorWithOffsets<'a> {
         Some(result)
     }
 }
+
+/// Trait implemented by types that can visit all [`Operator`] variants.
+#[allow(missing_docs)]
+pub trait VisitOperator<'a, Input> {
```

Comment:
Personally I still don't understand why there's an `Input` type parameter here. The source, `BinaryReader`, can only pass `usize` here. The other usage of `(usize, &T)` in the validator could instead be implemented with something like `impl VisitOperator for SomeTemporary<'a> { ... }` where the `&T` lives in the `SomeTemporary<'_>` or something like that.

I think you mentioned that there may be a perf difference? Can you link to two commits which differ just with this that I could help take a look?

Reply:
I implemented another potential solution that does not require an `Input` parameter [in this branch](https://github.com/Robbepop/wasm-tools/tree/rf-visit-operator-inner). However, I disliked the solution for 2 reasons.

1. It performed worse than the other solution and the original design.
2. It kinda felt like a hacky attempt to me whereas the other solution for which I went was just clean imo.

Reply:
Resolved, since I just implemented a design that got rid of the unwanted `Input` parameter.